src/embedding/kd.ts

```diff
@@ -0,0 +1,441 @@
+        IKdDrillableItemsCommandBody
+    >;
+
+    /**
+     * Data type of drillable items command
+     *
+     * Note: The main event data was wrapped to application and product data structure
+     * @see IKdDrillableItemsCommandBody
+     */
+    export type DrillableItemsCommandData = IKdGdcMessageEnvelope<
+        GdcKdCommandType.DrillableItems,
+        IKdDrillableItemsCommandBody
+    >;
+
+    /**
+     * Type-guard checking whether object is an instance of {@link DrillableItemsCommandData}.
+     *
+     * @param obj - object to test
+     */
+    export function isDrillableItemsCommandData(obj: any): obj is DrillableItemsCommandData {
+        return getEventType(obj) === GdcKdCommandType.DrillableItems;
+    }
+
+    export interface ISetSizeCommandBody {
+        /**
+         * the height of the hosting iframe
+         */
+        heigth: number;
+    }
+
+    export type SetSizeCommand = IKdMessageEvent<GdcKdCommandType.SetSize, ISetSizeCommandBody>;
+
+    export type SetSizeCommandData = IKdGdcMessageEnvelope<GdcKdCommandType.SetSize, ISetSizeCommandBody>;
+
+    /**
+     * Type-guard checking whether object is an instance of {@link SetSizeCommandData}.
+     *
+     * @param obj - object to test
+     */
+    export function isSetSizeCommandData(obj: any): obj is SetSizeCommandData {
+        return getEventType(obj) === GdcKdCommandType.SetSize;
+    }
+
+    export interface INoPermissionsBody {
+        /**
+         * the 'data' section contains information about whether view or edit permissions are missing
+         */
+        reason: string;
+    }
+
+    /**
+     * This event is emitted When User does not have permissions to view or edit the content
+     */
+    export type NoPermissionsEventData = IKdGdcMessageEnvelope<
+        GdcKdEventType.NoPermissions,
+        INoPermissionsBody
+    >;
+
+    export interface IResizedBody {
+        height: number;
+    }
+
+    /**
+     * This event is emitted when the content is fully loaded
+     */
+    export type ResizedEventData = IKdGdcMessageEnvelope<
+        GdcKdEventType.Resized,
+        IResizedBody
+    >;
+
+    export interface IDashboardObjectMeta {
+        /**
+         * Client id - Each client has an identifier unique within the domain
+         *
+         * Note: use the combination of the data product ID and client ID instead of the project ID
+         */
+        client?: string;
+
+        /**
+         * object id
+         */
+        dashboardId: string;
+        /**
+         * Project id
+         */
+        project: string;
+        /**
+         * dashboard identifier
+         */
+        dashboard: string;
+
+        /**
+         * dashboard title - this is what users see in KD top bar (if visible)
+         */
+        title: string;
+    }
+
+    export type IDashboardBody = IKdAvailableCommands & IDashboardObjectMeta;
+
+    /**
+     * Data type of event that was emit when a dashboard has been created and saved.
```

Comment:
`... was emitted ...` 
search and replace would be great :)

Reply:
updated